typing/ctype.ml

```diff
@@ -1235,7 +1235,7 @@ let rec copy ?partial ?keep_names scope ty =
               in
               (* Register new type first for recursion *)
               Private_type_expr.set_desc
-                more (Tsubst(newgenty(Ttuple[more';t])));
+                more (Tsubst (more', Some t));
```

Comment:
All code paths aware of the this tuple encoding (aka the pattern matching above and one in `Subst`) never use `more'`. Since all other users would erroneously interpret this encoding as a real tuple, it seems to me that rather than the product `type_expr * type_expr option`, the non-encoded argument of `Tsubst` is a sum.

Reply:
If we were to just encode the current behavior of the algorithm in this type, you are correct.
However, it seems cleaner that the meaning of `Tsubst` be regular, i.e. its first argument is always a copy of the original node. Then the second argument is some special information used by the algorithm, here a copy of the enclosing `Tvariant` node.
We can ensure that we do not depart from the current behavior of the algorithm by checking that this second argument is always `None` when we use the first one. We will add such assertions.